core/common/test/LocalTimeTest.kt

```diff
@@ -85,6 +83,58 @@ class LocalTimeTest {
         assertFailsWith<IllegalArgumentException> { LocalTime(0, 0, 0, 1_000_000_000) }
     }
 
+    @Test
+    fun fromNanosecondOfDay() {
+        val data = mapOf(
+            0L to LocalTime(0, 0),
+            5000000001L to LocalTime(0, 0, 5, 1),
+            44105000000100L to LocalTime(12, 15, 5, 100),
+            86399000000999L to LocalTime(23, 59, 59, 999),
+        )
+
+        data.forEach { (nanosecondOfDay, localTime) ->
+            assertEquals(nanosecondOfDay, localTime.nanosecondOfDay)
+            assertEquals(localTime, LocalTime.fromNanosecondOfDay(nanosecondOfDay))
+        }
+    }
+
+    @Test
+    fun fromNanosecondOfDayInvalid() {
+        LocalTime.fromNanosecondOfDay(0)
+        assertFailsWith<IllegalArgumentException> { LocalTime.fromNanosecondOfDay(-1) }
+        assertFailsWith<IllegalArgumentException> { LocalTime.fromNanosecondOfDay(Long.MAX_VALUE) }
```

Comment:
A good idea is to check the boundaries. For example, in the test above, in addition to (or instead of) testing `86399000000999L`, `86399999999999L`, the maximum admissible value, could be tested, and here we could test `86400000000000L`, the first non-admissible value.